pkg/service/group_service.go

```diff
@@ -106,3 +106,260 @@ func (svc *GroupService) DeleteGroup(id string) error {
+
+	if len(id) == 0 {
+		return nil, fmt.Errorf("error: invalid parameter specified id:%v, skipStartCheckMask:%v ", id, skipStartCheckMask)
+	}
+
+	resp, err := svc.objectSet.SoftwareUpdateStart(&id, skipStartCheckMask)
+	return resp, err
+
+}
+
+// SoftwareDownloadGroup - download software update package.
+//   Required parameters:
+//       id - ID of the group.
+//       version - Version string to download.
+
+//   Optional parameters:
+//       force - Flag to force download.
+
+func (svc *GroupService) SoftwareDownloadGroup(id string, version string, force *bool) error {
+
+	if len(id) == 0 || len(version) == 0 {
+		return fmt.Errorf("error: invalid parameter specified id:%v, version:%v, force:%v ", id, version, force)
+	}
+
+	err := svc.objectSet.SoftwareDownload(&id, &version, force)
+	return err
+}
+
+// SoftwareCancelDownloadGroup - cancel ongoing download of software.
+//   Required parameters:
+//       id - ID of the group.
+
+func (svc *GroupService) SoftwareCancelDownloadGroup(id string) error {
+
+	if len(id) == 0 {
+		return fmt.Errorf("error: invalid parameter specified id:%v ", id)
+	}
+
+	err := svc.objectSet.SoftwareCancelDownload(&id)
+	return err
+}
+
+// SoftwareUpdateResumeGroup - resume stopped software update.
+//   Required parameters:
+//       id - ID of the group.
+
+func (svc *GroupService) SoftwareUpdateResumeGroup(id string) error {
+
+	if len(id) == 0 {
+		return fmt.Errorf("error: invalid parameter specified id:%v ", id)
+	}
+
+	err := svc.objectSet.SoftwareUpdateResume(&id)
+	return err
+}
+
+// GetGroupDiscoveredListGroup - get list of discovered groups with arrays that are initialized.
+//   Required parameters:
+//       id - ID of the group.
+
+//   Optional parameters:
+//       groupName - Name of the group requested to be discovered.
+
+func (svc *GroupService) GetGroupDiscoveredListGroup(id string, groupName *string) (*nimbleos.NsDiscoveredGroupListReturn, error) {
+
+	if len(id) == 0 {
+		return nil, fmt.Errorf("error: invalid parameter specified id:%v, groupName:%v ", id, groupName)
+	}
+
+	resp, err := svc.objectSet.GetGroupDiscoveredList(&id, groupName)
+	return resp, err
+
+}
+
+// ValidateMergeGroup - perform group merge validation.
+//   Required parameters:
+//       id - ID of the group.
+//       srcGroupName - Name of the source group.
+//       srcGroupIp - IP address of the source group.
+//       srcUsername - Username of the source group.
+//       srcPassword - Password of the source group.
+
+//   Optional parameters:
+//       srcPassphrase - Source group encryption passphrase.
+//       skipSecondaryMgmtIp - Skip check for secondary management IP address.
+
+func (svc *GroupService) ValidateMergeGroup(id string, srcGroupName string, srcGroupIp string, srcUsername string, srcPassword string, srcPassphrase *string, skipSecondaryMgmtIp *bool) (*nimbleos.NsGroupMergeReturn, error) {
```

Comment:
Some of the function names get a bit difficult to parse.  I wonder, might it be better to remove the noun from the function name, since anyway it is implied by the structure it acts on?

Reply:
It will be a lot more complicated to handle such things in a generator.

Reply:
Oh, I was guessing that the generator was putting the verb and noun together.  If the name comes directly from the source, I agree it doesn't seem practical to add logic to break them apart.